### PR TITLE
refactor: refactor tag-to-digest provider based on latest external data changes

### PR DESCRIPTION
**Pull request by chewong**
- remove the use of vendoring
- bump mutation API to v1beta1 and change main.go accordingly
- bump go version to 1.17

Example:
```
kubectl apply -f policy/examples/test.yaml --dry-run=server -ojson | jq '.spec.template.spec.containers'
[
  {
    "image": "gcr.io/distroless/static:nonroot@sha256:80c956fb0836a17a565c43a4026c9c80b2013c83bea09f74fa4da195a59b7a99",
    "imagePullPolicy": "IfNotPresent",
    "name": "tag",
    "resources": {},
    "terminationMessagePath": "/dev/termination-log",
    "terminationMessagePolicy": "File"
  },
  {
    "image": "gcr.io/distroless/static:nonroot@sha256:c9f9b040044cc23e1088772814532d90adadfa1b86dcba17d07cb567db18dc4e",
    "imagePullPolicy": "IfNotPresent",
    "name": "digest",
    "resources": {},
    "terminationMessagePath": "/dev/termination-log",
    "terminationMessagePolicy": "File"
  },
  {
    "image": "busybox@sha256:caa382c432891547782ce7140fb3b7304613d3b0438834dce1cad68896ab110a",
    "imagePullPolicy": "Always",
    "name": "dockerhub-latest",
    "resources": {},
    "terminationMessagePath": "/dev/termination-log",
    "terminationMessagePolicy": "File"
  }
]
```

Provider log:

```
kubectl logs tagtodigest-provider-688fc659fb-f8dhj
2022-03-14T22:09:52.743Z        INFO    tagToDigest-provider/main.go:40 starting server...
2022-03-14T22:12:38.174Z        INFO    tagToDigest-provider/main.go:130        mutate  {"response": {"apiVersion":"externaldata.gatekeeper.sh/v1alpha1","kind":"ProviderResponse","response":{"idempotent":true,"items":[{"key":"busybox","value":"busybox@sha256:caa382c432891547782ce7140fb3b7304613d3b0438834dce1cad68896ab110a"},{"key":"gcr.io/distroless/static:nonroot","value":"gcr.io/distroless/static:nonroot@sha256:80c956fb0836a17a565c43a4026c9c80b2013c83bea09f74fa4da195a59b7a99"},{"key":"gcr.io/distroless/static:nonroot@sha256:c9f9b040044cc23e1088772814532d90adadfa1b86dcba17d07cb567db18dc4e","value":"gcr.io/distroless/static:nonroot@sha256:c9f9b040044cc23e1088772814532d90adadfa1b86dcba17d07cb567db18dc4e"}]}}}
```

### CODE REVIEW

**Review by sozercan — APPROVED** [review]

**Comment by sozercan** [comment]
@chewong looks like a lint issue with unused var

File: policy/provider.yaml [hunk 1]
```diff
@@ -3,7 +3,5 @@ kind: Provider
 metadata:
   name: tagtodigest-provider
 spec:
-  proxyURL: http://tagtodigest-provider.default:8090/mutate
+  url: http://tagtodigest-provider.default:8090/mutate
   timeout: 3
```
Comment (sozercan) [inline policy/provider.yaml:7]: since mwh timeout is 2 seconds now, should we decrease this?

File: policy/examples/test.yaml [hunk 2]
```diff
@@ -20,3 +20,5 @@ spec:
         image: gcr.io/distroless/static:nonroot
       - name: digest # already includes digest, should not mutate
         image: gcr.io/distroless/static:nonroot@sha256:c9f9b040044cc23e1088772814532d90adadfa1b86dcba17d07cb567db18dc4e
+      - name: dockerhub-latest
+        image: busybox
```
Comment (sozercan) [inline policy/examples/test.yaml:24]: 👍

File: policy/provider.yaml (hunk 1 above)
Comment (chewong) [reply policy/provider.yaml:7]: I still see the mutating webhook timeout being [3 seconds](https://github.com/open-policy-agent/gatekeeper/blob/master/manifest_staging/charts/gatekeeper/values.yaml#L19)

Comment (sozercan) [reply policy/provider.yaml:7]: chart is updated now, let's update this to 1

File: main.go [hunk 3]
```diff
@@ -23,10 +23,9 @@ import (
 var log logr.Logger
 
 const (
-	timeout      = 3 * time.Second
-	providerName = "tagtodigest-provider"
-	apiVersion   = "externaldata.gatekeeper.sh/v1alpha1"
-	kind         = "ProviderResponse"
+	timeout    = 3 * time.Second
```
Comment (sozercan) [inline main.go:26]: we should change this timeout too

**Review by sozercan — APPROVED** [review]

**Pull request merged**